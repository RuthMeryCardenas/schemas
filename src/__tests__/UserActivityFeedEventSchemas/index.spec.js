const mongoose = require('mongoose/browser');
const { UserActivityFeedEventSchemas: { UserActivityFeedEventSchema } } = require('../../../')(mongoose);


describe('UserActivityFeedEventSchema', () => {
  it('should fail validation when missing fields are provided', () => {
    const doc = new mongoose.Document({}, UserActivityFeedEventSchema);
    expect(doc.validateSync().errors).toMatchSnapshot();
  });

  it('should successfully validate with proper values', (done) => {
    const doc = new mongoose.Document({
      doc: mongoose.Types.ObjectId(),
      type: 'GenericEvent',
    }, UserActivityFeedEventSchema);

    return doc.validate((err) => {
      expect(err).toBe(null);
      done();
    });
  });
});
