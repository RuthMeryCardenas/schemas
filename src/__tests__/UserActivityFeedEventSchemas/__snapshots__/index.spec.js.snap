// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserActivityFeedEventSchema should fail validation when missing fields are provided 1`] = `
Object {
  "doc": [ValidatorError: Path \`doc\` is required.],
  "type": [ValidatorError: Path \`type\` is required.],
}
`;
